src/utils/config_utils/src/config_utils.c

```diff
@@ -448,3 +640,68 @@ void ADUC_ConfigInfo_FreeAduShellTrustedUsers(VECTOR_HANDLE users)
 
     VECTOR_clear(users);
 }
+
+/**
+ * @brief Get the existing ADUC_ConfigInfo object, or create one if it doesn't exist.
+ *
+ * @return const ADUC_ConfigInfo* a pointer to ADUC_ConfigInfo object. NULL if failure.
+ * Caller must call ADUC_ConfigInfo_Release to free the object.
+ */
+const ADUC_ConfigInfo* ADUC_ConfigInfo_GetInstance()
+{
+    const ADUC_ConfigInfo* config = NULL;
+
+    s_config_lock();
```

Comment:
Why lock and refCount is needed? Trying to save memory? Why can't it just read config into global obj once during startup on main thread? then do not need lock / refcount for multiple threads to read the data.  Is profiling show that it is large consumer of memory?

Reply:
The main purpose of this change is to ensure that the configuration file is parsed only once, as needed, per process. Furthermore, as you've mentioned, if the main thread creates the file once and frees it upon exit, the other threads will not need to parse the configuration file again.

Reply:
I see. I still think could call init and uninit in main func of each process, then a little simpler and don't need refcount / lock. refcount makes less sense here since refcount never expected to go to 0.

Reply:
As discussed off-line, this is a bit more practical when the agent forks many child processes that sometime need to load configs. Refcount make it simple to manages the config info lifecycle